java/compaction/compaction-job-creation/src/test/java/sleeper/compaction/job/creation/CreateCompactionJobsTest.java

```diff
@@ -38,272 +40,354 @@
+            stateStore2.addFiles(List.of(fileReference5, fileReference6, fileReference7, fileReference8));
+
+            // When
+            CreateCompactionJobs jobCreator = CreateCompactionJobs.standard(
+                    ObjectFactory.noUserJars(), instanceProperties,
+                    new FixedTablePropertiesProvider(List.of(tableProperties1, tableProperties2)),
+                    new FixedStateStoreProvider(Map.of(
+                            tableProperties1.getId().getTableName(), stateStore1,
+                            tableProperties2.getId().getTableName(), stateStore2)),
+                    jobs::add, jobStatusStore);
+            jobCreator.createJobs(tableProperties1);
+
+            // Then
+            assertThat(jobs).singleElement().satisfies(job -> {
+                assertThat(job).isEqualTo(CompactionJob.builder()
+                        .jobId(job.getId())
+                        .tableId(tableProperties1.get(TABLE_ID))
+                        .inputFiles(List.of("file1", "file2", "file3", "file4"))
+                        .outputFile(job.getOutputFile())
+                        .partitionId("root")
+                        .build());
+                assertThat(stateStore1.getFileReferences())
+                        .usingRecursiveFieldByFieldElementComparatorIgnoringFields("lastStateStoreUpdateTime")
+                        .containsExactly(
+                                withJobId(fileReference1, job.getId()),
+                                withJobId(fileReference2, job.getId()),
+                                withJobId(fileReference3, job.getId()),
+                                withJobId(fileReference4, job.getId()));
+                verifyJobCreationReported(job);
+            });
+            assertThat(stateStore2.getFileReferences())
+                    .containsExactly(fileReference5, fileReference6, fileReference7, fileReference8);
+        }
     }
 
-    private void setFileReferences(List<FileReference> fileReferences) throws Exception {
-        stateStore.addFiles(fileReferences);
-    }
+    @Nested
+    @DisplayName("Compact all files")
+    class CompactAllFiles {
+        private final List<CompactionJob> jobs = new ArrayList<>();
+        private final CreateCompactionJobs jobCreator = CreateCompactionJobs.compactAllFiles(
+                ObjectFactory.noUserJars(), instanceProperties,
+                new FixedTablePropertiesProvider(tableProperties),
+                new FixedStateStoreProvider(tableProperties, stateStore),
+                jobs::add, jobStatusStore);
 
-    private void verifySetJobForFilesInStateStore(String jobId, List<FileReference> fileReferences) {
-        assertThat(fileReferences).allSatisfy(fileReference ->
-                assertThat(getActiveStateFromStateStore(fileReference).getJobId()).isEqualTo(jobId));
-    }
+        @BeforeEach
+        void setUp() {
+            jobs.clear();
+        }
```

Comment:
Each test gets a new instance of the test class, so it'll be a fresh list every time. This can be removed.

Reply:
Done, thanks